src/test/clojure/clara/test_rules.clj

```diff
@@ -2361,7 +2361,22 @@
                   (query check-exception))
               (is false "Scope binding test is expected to throw an exception.")
               (catch Exception e e))]
+      (is (= {:?w 10, :?t nil}
+             (-> e (ex-data) :bindings))))))
+
+(deftest test-local-scope-visible-in-test-filter
+  (let [check-exception (assoc (dsl/parse-query [] [[WindSpeed (= ?w windspeed)]
+                                                    [Temperature (= ?t temperature)]
+                                                    [:test (> ?t ?w)]])
+                               :name "my-test-query")]
 
+    (let [e (try
+              (-> (mk-session [check-exception])
+                  (insert (->WindSpeed 10 "MCI") (->Temperature nil "MCI"))
+                  (fire-rules)
+                  (query check-exception))
+              (is false "Scope binding test is expected to throw an exception.")
+              (catch Exception e e))]
```

Comment:
Shouldn't there be some validation that the exception has the useful information that was added in this PR?

Reply:
Yes, I'll add additional assertions to validate that.